main/CMakeLists.txt

```diff
@@ -51,7 +51,7 @@ foreach(rawUtilName ${utils})
     # We need the .py only on Windows
     string(REPLACE ".py" "" utilName ${utilName})
   endif()
-  get_filename_component(python ${PYTHON_EXECUTABLE} NAME)
+  set(python ${PYTHON_EXECUTABLE})
```

Comment:
A proposal for the commit message:

```
The ROOT command line tools (e.g. rootls, rootmkdir, ...) are configured with
a header that tells the script which python to use. In the old approach, what
was written in that header was just the name of the Python binary file in
${PYTHON_EXECUTABLE}, while the path was stripped out. This led to an
issue with the brew installation on MacOs, since ROOT was built with a
python3 installation (3.8 to be more exact) and therefore "python3" was
injected in the header of the command line tools. However, the system
had another python3 installation (3.7), which was the one executed
when just running "python3". This caused that the command line tools
ran with a different python installation than the one ROOT was built
with, and this led to a crash (ROOT-10720).

This commit solves the issue by injecting the full path to the
(highest) python ROOT was built with in the header of the
command line tools.
```